sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -175,10 +175,11 @@ private[sql] case class DataSourceScan(
   }
 
   private def canProcessBatches(): Boolean = {
+    lazy val conf = SQLContext.getActive().get.conf
     relation match {
       case r: HadoopFsRelation if r.fileFormat.isInstanceOf[ParquetSource] &&
-        SQLContext.getActive().get.conf.getConf(SQLConf.PARQUET_VECTORIZED_READER_ENABLED) &&
-        SQLContext.getActive().get.conf.getConf(SQLConf.WHOLESTAGE_CODEGEN_ENABLED) =>
+        conf.parquetVectorizedReaderEnabled &&
+        conf.wholeStageEnabled && schema.length <= conf.wholeStageMaxNumFields =>
```

Comment:
We really need to move all of this logic into the planner.  I don't think it would be that hard after the refactoring we did.

I think main complication is that the parquet reader only supports a subset of possible schemata.  So you have to branch based both on the format but also arbitrary functions of the schema.